packages/leancode_contracts/lib/src/types/date_only.dart

```diff
@@ -4,7 +4,17 @@
 class DateOnly implements Comparable<DateOnly> {
   /// Constructs a [DateOnly] from `year`, `month`, and `day`.
   /// Additionally asserts that it represents a valid date.
-  const DateOnly(this.year, this.month, this.day) : assert(year >= 0);
+  DateOnly(this.year, this.month, this.day)
+      : assert(
```

Comment:
I'm wondering whether an assertion is enough. In the context of communication coming from the server, it's safe to trust the contract and that the values will always be valid. But what about user input, that in tests just wasn't caught, but it's possible to provide some invalid value somehow? Like a birthday inputs that accept 3 individual ints.

Should we throw here instead? What's your opinion on that?

Reply:
I think we should either throw or mimic the behavior of DateTime (overflowing to other components). Maybe we should even get rid of the assert, and just do overflowing like DateTime does. It might be useful for end users that want to implement "next day" "next month" etc logic. But then there is nothing warning the user that some date was incorrect. 

Are you leaning towards any solution? I definitely do not want to leave it as is and let the type represent invalid dates. Same issue applies to TimeOnly.

Reply:
I prefer accepting only a correct data and throw otherwise

Reply:
Changed to throwing errors